src/hotspot/share/ci/ciMethod.cpp

```diff
@@ -1142,6 +1144,28 @@ int ciMethod::code_size_for_inlining() {
 // Also some instructions inside the code are excluded from inline
 // heuristic (e.g. post call nop instructions; see InlineSkippedInstructionsCounter)
 int ciMethod::inline_instructions_size() {
+  if (_inline_instructions_size == -1) {
```

Comment:
Why repeat this condition and not put new code under existing one?

Reply:
The caching logic sets the `_inline_instructions_size` if the value is found in the archive. The normal logic doesn't need to run if this happens. 
Something like:
```
  if (_value == -1) {
       _value = get_from_cache();
  }
  if (_value == -1) { // didn't get it from the cache
    _value = compute_value();
  }
```
So basically to delineate the caching logic from the normal path. Does it make sense?